src/Jitter_Optimize.cpp

```diff
@@ -1129,6 +1130,78 @@ bool CJitter::ConstantPropagation(StatementList& statements)
 	return changed;
 }
 
+bool CJitter::CopyPropagation2(StatementList& statements)
+{
+	bool changed = false;
+
+	for(auto outerStatementIterator(statements.begin());
+		statements.end() != outerStatementIterator; ++outerStatementIterator)
+	{
+		STATEMENT& outerStatement(*outerStatementIterator);
+
+		//Some operations we can't propagate
+		if(outerStatement.op != OP_ADD) continue;
+
+		CSymbolRef* outerDstSymbol = outerStatement.dst.get();
+		if(outerDstSymbol == NULL) continue;
+
+		//Don't mess with relatives
+		if(outerDstSymbol->GetSymbol()->IsRelative())
+		{
+			continue;
+		}
+
```

Comment:
We should check if src2 is a constant. No point in continuing if we're not adding a constant.